distribution/pull_v2.go

```diff
@@ -15,26 +15,23 @@ import (
 	"github.com/docker/distribution/manifest/schema1"
 	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/distribution/metadata"
+	"github.com/docker/docker/distribution/xfer"
 	"github.com/docker/docker/image"
 	"github.com/docker/docker/image/v1"
 	"github.com/docker/docker/layer"
-	"github.com/docker/docker/pkg/archive"
-	"github.com/docker/docker/pkg/broadcaster"
-	"github.com/docker/docker/pkg/progressreader"
-	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/docker/docker/pkg/ioutils"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/registry"
 	"golang.org/x/net/context"
 )
 
 type v2Puller struct {
+	ctx            context.Context
```

Comment:
Tried it both ways, ended up preferring this one for being cleaner and less invasive. There are some other examples of a ctx being part of a struct that carries context in the distribution repo. The pull_v2 code doesn't split into concurrent goroutines so I don't think that's a concern.